PR: https://github.com/odoo/odoo/pull/253038

From: 083d9dbc640c422cb014f8f7e81399ab636d609c
From: Antoine Boonen
Date: 2026-04-30 15:29:46

Breaking data model changes scores: del:9 + add:0, change matches:
-    search_type = fields.Selection([('companies', 'Companies'), ('people', 'Companies and their Contacts')], string='Target', required=True, default='companies')
-    contact_number = fields.Integer(string='Number of Contacts', default=10)
-    contact_filter_type = fields.Selection([('role', 'Role'), ('seniority', 'Seniority')], string='Filter on', default='role')
-    preferred_role_id = fields.Many2one('crm.iap.lead.role', string='Preferred Role')
-    role_ids = fields.Many2many('crm.iap.lead.role', string='Other Roles')
-    seniority_id = fields.Many2one('crm.iap.lead.seniority', string='Seniority')
-    lead_contacts_credits = fields.Char(compute='_compute_tooltip', readonly=True)
-    lead_total_credits = fields.Char(compute='_compute_tooltip', readonly=True)

Total Changes: 53

[REM] crm_iap_mine,website_crm_iap_reveal : Remove 'person' lead

enrichement option

Problem
*--------
When we switched the provider from Clearbit to Dun & Bradstreet, we lost
the ability to import contacts when doing lead generation, as we didn't
have access to the data.
By checking what D&B has available, we realised that they can't give us
contact information other than generic info (like firstname, name),
because of GDPR.

Losing the direct contact information undermines quite a lot the value
of the service, so instead of figuring a complicated, probably non-GDPR
way of getting that information back, we'll move the contact part of
the Lead Generation into `website_crm_iap_reveal` for Clearbit usage
only.

task-5734814

closes odoo/odoo#253038

Related: odoo/upgrade#9650
Signed-off-by: Claire Bretton (clbr) <clbr@odoo.com>

================================= pseudo patch: =================================

--- a/addons/crm_iap_mine/models/crm_iap_lead_helpers.py
+++ b/addons/crm_iap_mine/models/crm_iap_lead_helpers.py
@@ -29,7 +29,7 @@ class CrmIapLeadHelpers(models.Model):
         self.env['ir.config_parameter'].sudo().set_bool(notification_parameter, True)
 
     @api.model
-    def lead_vals_from_response(self, lead_type, team_id, tag_ids, user_id, company_data, people_data):
+    def lead_vals_from_response(self, lead_type, team_id, tag_ids, user_id, company_data):
         country_id = company_data.get('country_id')
         if not country_id:
             country_id = self.env['res.country'].search([('code', '=', company_data['country_code'])]).id
@@ -54,14 +54,6 @@ class CrmIapLeadHelpers(models.Model):
             'country_id': country_id,
             'state_id': self._find_state_id(company_data.get('state_code'), country_id),
         }
-
-        # If type is people then add first contact in lead data
-        if people_data:
-            lead_vals.update({
-                'contact_name': people_data[0]['full_name'],
-                'email_from': people_data[0]['email'],
-                'function': people_data[0]['title'],
-            })
         return lead_vals
 
     @api.model

--- a/addons/crm_iap_mine/models/crm_iap_lead_mining_request.py
+++ b/addons/crm_iap_mine/models/crm_iap_lead_mining_request.py
@@ -14,8 +14,6 @@ DEFAULT_ENDPOINT = 'https://iap-services.odoo.com'
 
 MAX_LEAD = 200
 
-MAX_CONTACT = 5
-
 CREDIT_PER_COMPANY = 1
 CREDIT_PER_CONTACT = 1
 
@@ -38,7 +36,6 @@ class CrmIapLeadMiningRequest(models.Model):
 
     # Request Data
     lead_number = fields.Integer(string='Number of Leads', required=True, default=3)
-    search_type = fields.Selection([('companies', 'Companies'), ('people', 'Companies and their Contacts')], string='Target', required=True, default='companies')
     error_type = fields.Selection([
         ('credits', 'Insufficient Credits'),
         ('no_result', 'No Result'),
@@ -64,35 +61,18 @@ class CrmIapLeadMiningRequest(models.Model):
     available_state_ids = fields.One2many('res.country.state', compute='_compute_available_state_ids')
     industry_ids = fields.Many2many('crm.iap.lead.industry', string='Industries')
 
-    # Contact Generation Filter
-    contact_number = fields.Integer(string='Number of Contacts', default=10)
-    contact_filter_type = fields.Selection([('role', 'Role'), ('seniority', 'Seniority')], string='Filter on', default='role')
-    preferred_role_id = fields.Many2one('crm.iap.lead.role', string='Preferred Role')
-    role_ids = fields.Many2many('crm.iap.lead.role', string='Other Roles')
-    seniority_id = fields.Many2one('crm.iap.lead.seniority', string='Seniority')
-
     # Fields for the blue tooltip
     lead_credits = fields.Char(compute='_compute_tooltip', readonly=True)
-    lead_contacts_credits = fields.Char(compute='_compute_tooltip', readonly=True)
-    lead_total_credits = fields.Char(compute='_compute_tooltip', readonly=True)
 
-    @api.depends('lead_number', 'contact_number')
+    @api.depends('lead_number')
     def _compute_tooltip(self):
         for record in self:
             company_credits = CREDIT_PER_COMPANY * record.lead_number
-            contact_credits = CREDIT_PER_CONTACT * record.contact_number
-            total_contact_credits = contact_credits * record.lead_number
-            record.lead_contacts_credits = _(
-                "Up to %(credit_count)d additional credits will be consumed to identify %(contact_count)d contacts per company.",
-                credit_count=contact_credits * company_credits,
-                contact_count=record.contact_number,
-            )
             record.lead_credits = _(
                 "%(credit_count)d credits will be consumed to find %(company_count)d companies.",
                 credit_count=company_credits,
                 company_count=record.lead_number,
             )
-            record.lead_total_credits = _("This makes a total of %d credits for this request.", total_contact_credits + company_credits)
 
     @api.depends('lead_ids.lead_mining_request_id')
     def _compute_lead_count(self):
@@ -157,13 +137,6 @@ class CrmIapLeadMiningRequest(models.Model):
         elif self.lead_number > MAX_LEAD:
             self.lead_number = MAX_LEAD
 
-    @api.onchange('contact_number')
-    def _onchange_contact_number(self):
-        if self.contact_number <= 0:
-            self.contact_number = 1
-        elif self.contact_number > MAX_CONTACT:
-            self.contact_number = MAX_CONTACT
-
     @api.onchange('country_ids')
     def _onchange_country_ids(self):
         self.state_ids = []
@@ -198,7 +171,7 @@ class CrmIapLeadMiningRequest(models.Model):
         self.ensure_one()
         payload = {
             'lead_number': self.lead_number,
-            'search_type': self.search_type,
+            'search_type': 'companies',
             'countries': [{
                 'code': country.code,
                 'states': self.state_ids.filtered(lambda state: state in country.state_ids).mapped('code'),
@@ -218,14 +191,6 @@ class CrmIapLeadMiningRequest(models.Model):
                 for reveal_id in reveal_ids.split(',')
             ]
             payload['industry_ids'] = all_industry_ids
-        if self.search_type == 'people':
-            payload.update({'contact_number': self.contact_number,
-                            'contact_filter_type': self.contact_filter_type})
-            if self.contact_filter_type == 'role':
-                payload.update({'preferred_role': self.preferred_role_id.reveal_id,
-                                'other_roles': self.role_ids.mapped('reveal_id')})
-            elif self.contact_filter_type == 'seniority':
-                payload['seniority'] = self.seniority_id.reveal_id
         return payload
 
     def _perform_request(self):
@@ -279,7 +244,6 @@ class CrmIapLeadMiningRequest(models.Model):
             template_values = data
             template_values.update({
                 'flavor_text': _("Opportunity created by Odoo Lead Generation"),
-                'people_data': data.get('people_data'),
                 'country': country.name,
                 'zip_code': data.get('zip'),
                 'country_id': country.id,
@@ -299,8 +263,7 @@ class CrmIapLeadMiningRequest(models.Model):
     def _lead_vals_from_response(self, data):
         self.ensure_one()
         company_data = data
-        people_data = []
-        lead_vals = self.env['crm.iap.lead.helpers'].lead_vals_from_response(self.lead_type, self.team_id.id, self.tag_ids.ids, self.user_id.id, company_data, people_data)
+        lead_vals = self.env['crm.iap.lead.helpers'].lead_vals_from_response(self.lead_type, self.team_id.id, self.tag_ids.ids, self.user_id.id, company_data)
         lead_vals['lead_mining_request_id'] = self.id
         return lead_vals
 
